native_store.go

```diff
@@ -29,10 +31,18 @@ const (
 	emptyUsername           = "<token>"
 )
 
+// dockerCredentials mimics how docker credential helper binaries store
+// credential information.
+type dockerCredentials struct {
+	ServerURL string
+	Username  string
+	Secret    string
+}
+
 // nativeStore implements a credentials store using native keychain to keep
 // credentials secure.
 type nativeStore struct {
-	programFunc client.ProgramFunc
+	executer.Executer
```

Comment:
You need to give this component name like
```suggestion
	exec executer.Executer
```
Otherwise, all public method of `executer.Executer` will be exposed to the `nativeStore`. In other words, some one can do
```go
store := NewNativeStore("foobar")
exec := store.(executer.Executer)
exec.Execute(ctx, "hello", "get")
```
which should not be allowed.

Reply:
Added back the variable name.